snuba/datasets/plans/storage_builder.py

```diff
@@ -92,112 +98,78 @@ def get_query_data_source(
-        # The translation mappers to be used when translating the logical query
-        # into the clickhouse query.
-        self.__mappers = mappers if mappers is not None else TranslationMappers()
-        # This is a set of query processors that have to be executed on the
-        # query after the storage selection but that are defined by the dataset.
-        # Query processors defined by a Storage must be executable independently
-        # from the context the Storage is used (whether the storage is used by
-        # itself or whether it is joined with another storage).
-        # In a joined query we would have processors defined by multiple storages.
-        # that would have to be executed only once (like Prewhere). That is a
-        # candidate to be added here as post process.
-        self.__post_processors = post_processors or []
-
-    @with_span()
-    def build_and_rank_plans(
-        self, query: LogicalQuery, settings: QuerySettings
-    ) -> Sequence[ClickhouseQueryPlan]:
-        with sentry_sdk.start_span(
-            op="build_plan.single_storage", description="translate"
-        ):
-            # The QueryTranslator class should be instantiated once for each call to build_plan,
-            # to avoid cache conflicts.
-            clickhouse_query = QueryTranslator(self.__mappers).translate(query)
-
-        with sentry_sdk.start_span(
-            op="build_plan.single_storage", description="set_from_clause"
-        ):
-            clickhouse_query.set_from_clause(
-                get_query_data_source(
-                    self.__storage.get_schema().get_data_source(),
-                    final=query.get_final(),
-                    sampling_rate=query.get_sample(),
-                )
-            )
-
-        cluster = self.__storage.get_cluster()
-
-        db_query_processors = [
-            *self.__storage.get_query_processors(),
-            *self.__post_processors,
-            MandatoryConditionApplier(),
-            MandatoryConditionEnforcer(
-                self.__storage.get_mandatory_condition_checkers()
-            ),
-        ]
-
-        return [
-            ClickhouseQueryPlan(
-                query=clickhouse_query,
-                plan_query_processors=[],
-                db_query_processors=db_query_processors,
-                storage_set_key=self.__storage.get_storage_set_key(),
-                execution_strategy=SimpleQueryPlanExecutionStrategy(
-                    cluster=cluster,
-                    db_query_processors=db_query_processors,
-                    splitters=self.__storage.get_query_splitters(),
-                ),
-            )
-        ]
-
-
-class SelectedStorageQueryPlanBuilder(ClickhouseQueryPlanBuilder):
+class StorageQueryPlanBuilder(ClickhouseQueryPlanBuilder):
     """
-    A query plan builder that selects one of multiple storages in the dataset.
+    Builds the Clickhouse Query Execution Plan for a dataset which supports single storages,
```

Comment:
are there tests for all these branches of functionality? I understand that the existing tests pass but those depend on existing entities and storages right? Have we tested this component just in isolation?

Reply:
There are individual tests for storage selection and cluster selection. Both of which have been updated to use the consolidated plan. However, there is no test for just the component in isolation. I can make one for that!